### DISCUSSION TITLE
how to get orientations extracted from ODF in correct crystal symmetry

### DISCUSSION THREAD

**Discussion by YvesFl** [mtex-toolbox/mtex#2242] [Ask Anything] [answered]
Dear all,

I can import an ODF with a cubic crystal symmetry from a text file using the following function
 "odf = SO3Fun.load(fname,'cs',cs,'ss',ss,'resolution',res,'Bunge', 'ColumnNames',{'Euler 1','Euler 2','Euler 3','weights'});"
The Euler angles in the ODF text file go from 0, 0, 0 to 360, 90, 90 for phi1, Phi and phi2.
When trying to extract the 6 most important orientations using the function
"[value,ori] = max(odf,'numLocal',6)"
I get orientations with Euler angles where phi2 is above 90°. Hence I guess, the crystal symmetry does not seem to be considered in the calculation. 
Is there a possibility to get the orientations with Euler angles restricted to the angular space defined in the initial ODF or am I doing this the wrong way?
Thank you for your help.

Yves F.
![Screenshot_6-orientations](https://github.com/user-attachments/assets/6ae39ced-fb1e-4e5c-bf1e-a2432bf79ae4)
[odf_A02_c.txt](https://github.com/user-attachments/files/17293888/odf_A02_c.txt)

**Comment by ralfHielscher** [comment, marked as answer]
You are doing everything right. Just your conclusion, that MTEX does not consider crystal symmetry is wrong. In fact, to each of the orientations computed by MTEX there are **three** symmetrically equivalent orientations within the range of [0,360] x [0,90] x [0,90].
Which of them do you want to have? And why? You might want to consider the command
```matlab
[phi1,Phi,phi2] = ori.project2EulerFR
```

Ralf.

**Reply by YvesFl** [reply]
Dear Ralf,
Thank you for your response. This works for me.
Yves

**Comment by zmichels** [comment]
I'm not sure if this will help, and I'm not by my laptop to try... but maybe try (?):
project2FundamentalRegion()

Some info here:
https://mtex-toolbox.github.io/OrientationFundamentalRegion.html
